src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -84,10 +84,15 @@ internal static void VerifyIOperationFeatureFlag(bool isIOperationFeatureEnabled
 
         private static void VerifyDiagnosticLocationInCompilation(string id, Location location, Compilation compilation)
         {
-            if (location.IsInSource && !compilation.ContainsSyntaxTree(location.SourceTree))
+            if (!location.IsInSource)
+            {
+                return;
```

Comment:
yes. we have other unit test covering this part. also I added new unit test that cover new code path explicitly.